mindsdb_native/libs/helpers/locking.py

```diff
@@ -1,34 +1,77 @@
 import os
-
+import sys
+import inspect
+import traceback
 import portalocker
 
 from mindsdb_native.config import CONFIG
 
 
-def learn_lock(name):
-    learn_lock_fn = os.path.join(CONFIG.MINDSDB_STORAGE_PATH,f'{name}_learn.lock')
-    f = open(learn_lock_fn, 'a+')
-    portalocker.lock(f, portalocker.LOCK_EX+portalocker.LOCK_NB)
-    return f
-
-def predict_lock(name):
-    learn_lock_fn = os.path.join(CONFIG.MINDSDB_STORAGE_PATH,f'{name}_learn.lock')
-    f = open(learn_lock_fn, 'a+')
-    portalocker.lock(f, portalocker.LOCK_SH+portalocker.LOCK_NB)
-    return f
-
-def delete_lock(name):
-    learn_lock(name)
-    delete_lock_fn = os.path.join(CONFIG.MINDSDB_STORAGE_PATH,f'{name}_delete.lock')
-    f = open(delete_lock_fn, 'a+')
-    portalocker.lock(f, portalocker.LOCK_EX+portalocker.LOCK_NB)
-    return f
-
-def get_data_lock(name):
-    delete_lock_fn = os.path.join(CONFIG.MINDSDB_STORAGE_PATH,f'{name}_delete.lock')
-    f = open(delete_lock_fn, 'a+')
-    portalocker.lock(f, portalocker.LOCK_SH+portalocker.LOCK_NB)
-    return f
-
-def unlock(file):
-    portalocker.unlock(file)
+def mdb_lock(flags, lock_name, argname=None):
```

Comment:
Can you make the flag one of the two options (strings) `shared` and `exclusive`

Then add up the portalocker flags in here (SH + NB for shared, EX + NB for exclusive)

We always want non blocking locks so there's no need to specify that.

Reply:
done